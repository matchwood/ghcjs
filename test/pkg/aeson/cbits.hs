{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE OverloadedStrings #-}

module Main(main) where

import GHC.Generics
import Data.Aeson
import Data.Text


data Person = Person {
      name :: Text
    , age  :: Int
    } deriving (Generic, Show, Eq)

instance ToJSON Person where
    toEncoding = genericToEncoding defaultOptions

instance FromJSON Person where

main :: IO ()
main = do
  let p = Person {name = "Joe", age = 12}
  let enc = encode p
  putStrLn $ show enc
  case eitherDecode' enc of
    Left err -> putStrLn err
    Right np ->
      if np == p
        then putStrLn "Equal"
        else putStrLn "Not Equal"
