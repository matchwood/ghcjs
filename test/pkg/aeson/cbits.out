"{\"name\":\"Joe\",\"age\":12}"
Equal